plugins/faustwp/includes/rest/callbacks.php

```diff
@@ -202,6 +290,19 @@ function rest_telemetry_permission_callback( \WP_REST_Request $request ) {
 	return rest_authorize_permission_callback( $request );
 }
 
+/**
+ * Callback to check permissions for requests to `faustwp/v1/process_telemetry`.
+ *
+ * Authorized if the 'secret_key' settings value and http header 'x-faustwp-secret' match.
+ *
+ * @param \WP_REST_Request $request The current \WP_REST_Request object.
+ *
+ * @return bool True if current user can, false if else.
+ */
+function rest_process_telemetry_permission_callback( \WP_REST_Request $request ) {
+	return rest_authorize_permission_callback( $request );
+}
```

Comment:
@TeresaGobble When we save a user's preferences for their telemetry opt-in, we'll also want to generate a unique id for them (something like a UUID) as GA requires this. You can take a look at our previous implementation in the CLI for inspiration on how this can be done.